src/openpgp/packet/signature.iced

```diff
@@ -318,8 +320,33 @@ class Signature extends Packet
           if @issuer_matches_key(@primary)
             @primary.mark_revoked sig
           else
-            iki = @get_issuer_key_id()
-            err = new Error "can't revoke key ID #{iki.toString('hex')} (!= #{@primary.get_key_id().toString('hex')})"
+            @primary.add_designated_revocation sig
+
+        when T.direct
+          if fp = @subpacket_index.hashed[S.revocation_key]
+            @primary.add_designee fp
+
+        when T.certificate_revocation
+          if (userid = @data_packets[1].to_userid())?
+            userid.mark_revoked sig
+
+        else
+          err = new Error "Got unknown signature type=#{@type}"
```

Comment:
error here ends up as a warning: https://github.com/keybase/kbpgp/blob/master/src/openpgp/processor.iced#L135-L136

but I can remove the `else` block